content/questions/reduce-array/index.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Reduce Array
+tags:
+  - reduce
+  - array
+order: 39
+date: '2019-09-22'
+answers:
+  - 'hello // correct'
+  - 'eohll'
+  - 'Order not guaranteed'
+---
+
+Consider the following code block. What gets logged?
+
+```javascript
+const scrambled = { 2: "e", 5: "o", 1: "h", 4: "l", 3: "l" };
+
+const result = Object
+  .values(scrambled)
+  .reduce((agg, el) => agg + el, "");
+
+console.log(result);
+```
+
+<!-- explanation -->
+
+The reduce array method takes two arguments: a reducer function and an initial value. 
```

Comment:
Can you additionally add that `Object.values` iterates through enumerable number keys in sequential order? This is why the letters appear to sort themselves correctly.

Reply:
No problem! Done!